libtextsecure/message_receiver.js

```diff
@@ -120,7 +120,23 @@ MessageReceiver.prototype.extend({
       );
     }
 
-    const ev = new Event('message');
+    const ourNumber = textsecure.storage.user.getNumber();
+    const primaryDevice = window.storage.get('primaryDevicePubKey');
+    const isOurDevice =
+      message.source &&
+      (message.source === ourNumber || message.source === primaryDevice);
+    const isPublicChatMessage =
+      message.message.group &&
+      message.message.group.id &&
+      !!message.message.group.id.match(/^publicChat:/);
+    let ev;
+
+    if (isPublicChatMessage && isOurDevice) {
+      // Public chat messages from ourselves should be outgoing
+      ev = new Event('sent');
+    } else {
+      ev = new Event('message');
+    }
```

Comment:
trigger the 'sent'=outgoing event rather than the 'message'=incoming when receiving a public message from ourself so the logic is correct afterwards.